src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -228,8 +216,20 @@ private MongoClusterType determineClusterType() {
             return MongoClusterType.UnknownTarget;
         }
 
+        logger.log(Level.FINE, buildInfoRes.toString());
+
+        this.serverVersion = buildInfoRes.getFullVersion();
+
+        try {
+            this.serverMajorVersion = buildInfoRes.getMajorVersion();
+            this.serverMinorVersion = buildInfoRes.getMinorVersion();
+        } catch (IndexOutOfBoundsException e) {
+            logger.log(Level.SEVERE, e.toString());
+        }
```

Comment:
If versionArray can be null, you are not protected against Null Pointer Exception. 
Maybe use a catch all here.

```suggestion
        } catch (Exception e) {
            // Only log issues happening while trying to compute the server version as this is not a blocker.
            logger.log(Level.SEVERE, e.toString());
        }
```

Reply:
I'll add the comment. I can check for null pointer exception, but the versionArray has been returned as part of buildinfo since [4.4](https://www.mongodb.com/docs/v4.4/reference/command/buildInfo/). I stuck with the Index exception in case the return is malformed, though that seems like it'd be an exceedingly rare occurence.

Reply:
The indexOutOfBound is necessary because you throw. I just think that not being too specific in this case is good, but I agree we should never get there. It's up to you.